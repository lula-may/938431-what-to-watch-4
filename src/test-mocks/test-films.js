const movieTitles = [
  `Gone With The Wind`,
  `The Godfather`,
  `Casablanca`,
  `A Streetcar Named Desire`,
  `Some Likes It Hot`,
];

const movies = movieTitles.map((title, i) => {
  return {
    title,
    id: i,
    src: `poster-${i}.jpg`
  };
});

const headerMovie = {
  title: `Jurassic Park`,
  genre: `Sci-Fi`,
  releaseYear: 1993,
  bg: `background.jpg`,
  poster: `header-poster.jpg`
};

export {movies, headerMovie};

