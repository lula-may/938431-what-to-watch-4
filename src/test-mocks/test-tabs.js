const tabs = [
  {
    id: `aa`,
    type: `First`
  },
  {
    id: `bb`,
    type: `Second`,
  },
  {
    id: `cc`,
    type: `Third`,
  }
];

export {tabs};
