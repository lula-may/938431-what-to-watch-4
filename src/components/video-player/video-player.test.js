import React from "react";
import renderer from "react-test-renderer";
import VideoPlayer from "./video-player.jsx";
import {testMovies} from "../../test-mocks/test-films";

const mockSettings = {
  areControlsShown: false,
  height: `100`,
  isMuted: true,
  width: `200`,
};

const movie = testMovies[0];

it(`should correnctly render VideoPlayer component`, () => {
  const tree = renderer.create(
      <VideoPlayer
        isPlaying={false}
        settings={mockSettings}
        poster={movie.poster}
        src={movie.src}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
