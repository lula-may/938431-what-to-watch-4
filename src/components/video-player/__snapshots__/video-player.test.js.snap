// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correnctly render VideoPlayer component 1`] = `
<video
  height="100"
  poster="poster-0.jpg"
  width="200"
/>
`;
